crates/calibration/src/residuals.rs

```diff
@@ -0,0 +1,107 @@
+use std::f32::consts::FRAC_PI_2;
+
+use nalgebra::{DVector, Dyn, Owned, Vector};
+use types::{CameraPosition, FieldDimensions};
+
+use crate::{corrections::Corrections, lines::LinesError, measurement::Measurement};
+
+pub type Residual = Vector<f32, Dyn, ResidualStorage>;
+pub type ResidualStorage = Owned<f32, Dyn>;
+
+pub fn calculate_residuals_from_parameters(
+    parameters: &Corrections,
+    measurements: &[Measurement],
+    field_dimensions: &FieldDimensions,
+) -> Option<Residual> {
+    measurements
+        .iter()
+        .fold(
+            Some(Vec::new()),
+            |residuals: Option<Vec<f32>>, measurements| match residuals {
+                Some(mut residuals) => {
+                    residuals.extend::<Vec<f32>>(
+                        Residuals::calculate_from(parameters, measurements, field_dimensions)
+                            .ok()?
+                            .into(),
+                    );
+                    Some(residuals)
+                }
+                None => None,
+            },
+        )
+        .map(DVector::from_vec)
```

Comment:
I like iterators too, but in this case the procedural approach is a lot more readable:
```rs
    let mut residuals = Vec::new();
    for measurement in measurements {
        let vec: Vec<f32> = Residuals::calculate_from(parameters, measurement, field_dimensions)
            .ok()?
            .into();
        residuals.extend(vec.iter());
    }

    Some(DVector::from_vec(residuals));
```

Reply:
I agree with @knoellle  here

Reply:
Used and improved your suggestion.